common/src/test/java/org/opensearch/ml/common/connector/ConnectorActionTest.java

```diff
@@ -27,130 +26,124 @@
+            () -> new ConnectorAction(TEST_ACTION_TYPE, null, URL, null, TEST_REQUEST_BODY, null, null)
+        );
+        assertEquals("method can't be null", exception.getMessage());
     }
 
     @Test
     public void writeTo_NullValue() throws IOException {
-        ConnectorAction.ActionType actionType = ConnectorAction.ActionType.PREDICT;
-        String method = "http";
-        String url = "https://test.com";
-        ConnectorAction action = new ConnectorAction(actionType, method, url, null, null, null, null);
+        ConnectorAction action = new ConnectorAction(TEST_ACTION_TYPE, TEST_METHOD_HTTP, URL, null, TEST_REQUEST_BODY, null, null);
         BytesStreamOutput output = new BytesStreamOutput();
         action.writeTo(output);
         ConnectorAction action2 = new ConnectorAction(output.bytes().streamInput());
-        Assert.assertEquals(action, action2);
+        assertEquals(action, action2);
     }
 
     @Test
     public void writeTo() throws IOException {
-        ConnectorAction.ActionType actionType = ConnectorAction.ActionType.PREDICT;
-        String method = "http";
-        String url = "https://test.com";
         Map<String, String> headers = new HashMap<>();
         headers.put("key1", "value1");
-        String requestBody = "{\"input\": \"${parameters.input}\"}";
         String preProcessFunction = MLPreProcessFunction.TEXT_DOCS_TO_OPENAI_EMBEDDING_INPUT;
         String postProcessFunction = MLPostProcessFunction.OPENAI_EMBEDDING;
 
         ConnectorAction action = new ConnectorAction(
-            actionType,
-            method,
-            url,
+            TEST_ACTION_TYPE,
+            TEST_METHOD_HTTP,
+            URL,
             headers,
-            requestBody,
+            TEST_REQUEST_BODY,
             preProcessFunction,
             postProcessFunction
         );
         BytesStreamOutput output = new BytesStreamOutput();
         action.writeTo(output);
         ConnectorAction action2 = new ConnectorAction(output.bytes().streamInput());
-        Assert.assertEquals(action, action2);
+        assertEquals(action, action2);
     }
 
     @Test
     public void toXContent_NullValue() throws IOException {
-        ConnectorAction.ActionType actionType = ConnectorAction.ActionType.PREDICT;
-        String method = "http";
-        String url = "https://test.com";
-        ConnectorAction action = new ConnectorAction(actionType, method, url, null, null, null, null);
+        ConnectorAction action = new ConnectorAction(TEST_ACTION_TYPE, TEST_METHOD_HTTP, URL, null, TEST_REQUEST_BODY, null, null);
 
         XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
         action.toXContent(builder, ToXContent.EMPTY_PARAMS);
         String content = TestHelper.xContentBuilderToString(builder);
-        Assert.assertEquals("{\"action_type\":\"PREDICT\",\"method\":\"http\",\"url\":\"https://test.com\"}", content);
+        String expctedContent = """
```

Comment:
This doesn't support in Java 17. In main branch, we use JAVA 21 which support this string. So for you I would suggest to do:

1. Fix main branch code with not using `"""`
2. And then backport both of the commits together in 2.x branch.